src/openrct2-ui/windows/ServerList.cpp

```diff
@@ -683,29 +683,29 @@ static void fetch_servers()
         sort_servers();
     }
 
-    http_request_t request = {};
+    http::Request request;
     request.url = masterServerUrl;
-    request.method = HTTP_METHOD_GET;
-    request.body = nullptr;
-    request.type = HTTP_DATA_JSON;
+    request.method           = http::Method::GET;
+    request.header["Accept"] = "application/json";
     status_text = STR_SERVER_LIST_CONNECTING;
-    http_request_async(&request, fetch_servers_callback);
+    http::DoAsync(request, fetch_servers_callback);
 #endif
 }
 
 #ifndef DISABLE_HTTP
-static void fetch_servers_callback(http_response_t* response)
+static void fetch_servers_callback(http::Response response)
```

Comment:
This should pass response by reference (const reference if possible). Often I delete the copy constructor of a class to check I am not copying unnecessarily.